Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/VertexArrayMtl.mm

```diff
@@ -1101,7 +1105,7 @@ inline void SetDefaultVertexBufferLayout(mtl::VertexBufferLayoutDesc *layout)
         static_cast<VertexConversionBufferMtl *>(conversion);
     params.srcBuffer            = srcBuffer->getCurrentBuffer();
     params.srcBufferStartOffset = static_cast<uint32_t>(
-        MIN(static_cast<GLintptr>(vertexConversion->offset), binding.getOffset()));
```

Comment:
I think this existing `min` operation is dubious.
E.g. if the intention is to interpret the offset as uint32_t, then the min should operate on that, not that it first operates on int32_t and then converts to its inverse. 

Of course depends on whether the algorithm using srBufferStartOffset takes it into account that the srcBufferStartOffset is uint32_t but expects the 2s complement minus value to wrap around correctly and actually decrease the value it is being added to..

Reply:
Got it.  Looks to me like `srcBufferStartOffset` is used as a `uint32_t` everywhere.